### ISSUE TITLE
Show locale selector on login page

### ISSUE THREAD

**Issue by garethbowen** [medic/cht-core#6280]
**What feature do you want to improve?**
Currently on first run we show the login page in the instance default locale because we don't know what user is looking at the page yet. Once they've logged in we show a modal in the webapp to pick a language. If they change the language then we update the user settings doc so it'll be persisted for future sessions. Because the user settings doc is updated it needs to be replicated down which then prompts the user to refresh to get any changes. This is not a nice user experience.

**Describe the improvement you'd like**
If we give users a way to set their language before logging in then we can update the user settings doc before they've replicated it and do away with the in app modal _and_ the update app modal.

**Describe alternatives you've considered**
Alternatively you could leave the modal where it is and attempt to detect when the user settings doc hadn't changed in any important way, but this is more technically complex and only removes one of the two modals.

**Additional context**
Once #5954 is done then it'll be possible to translate the login page dynamically when the user changes their language which will help if they don't understand the default language.

**Comment by garethbowen** [comment]
Low priority because it only affects first run experience.

**Comment by garethbowen** [comment]
@michaelkohn @n-orlowski I prototyped a solution to this while stuck in an airport somewhere and would love some feedback on if you think this is an improvement on the current workflow and if there are further improvements that could be made.

### Current process

1. Login

![Screenshot from 2020-03-16 13-17-40](https://user-images.githubusercontent.com/2064938/76713928-05943180-6789-11ea-97cb-09b9057154cc.png)

2. Select Hindi from the modal dialogue

![Screenshot from 2020-03-16 13-19-04](https://user-images.githubusercontent.com/2064938/76713929-06c55e80-6789-11ea-895f-61ecc4ceadc4.png)

3. See tour options

![Screenshot from 2020-03-16 13-19-18](https://user-images.githubusercontent.com/2064938/76713930-075df500-6789-11ea-9c65-cfa38807861b.png)

### Prototype

1. Select language and login

![Screenshot from 2020-03-16 13-11-53](https://user-images.githubusercontent.com/2064938/76713965-35433980-6789-11ea-9b2f-f2e16bfb22fe.png)

2. Go straight to the tour select modal with your selected language

![Screenshot from 2020-03-16 13-16-01](https://user-images.githubusercontent.com/2064938/76713966-36746680-6789-11ea-8e05-f0e086e34e1c.png)

I think this is a better process because...

1. There is one fewer modal dialogue.
2. The app loading messages are translated into the user's selected language rather than default.
3. Once #5954 is done even the login page elements will be able to be translated to the user's language pre-login.
4. Once #6281 is done there will likely only be a couple of options to choose from rather than the eight shown in the screenshot above. If there is only one available language then the selector isn't shown.

One downside is if the user forgets to select a language then it's more difficult to find out how to set the language in the app, however this applies to if the user dismisses the dialogue in the current implementation.

**Comment by n-orlowski** [comment]
Cool! I like the idea of one less modal but do think that it seems easier for a user to miss the selection as it's below the login CTA. A couple of solutions could be to include languages in a dropdown above the button or if there are less than 3 options then maybe something like this

![image](https://user-images.githubusercontent.com/59581790/76777457-745f9380-6765-11ea-9a39-0821b827e190.png)

that could also update the language of the login text fields and button as an initial demonstration of its function

**Comment by garethbowen** [comment]
> that could also update the language of the login text fields and button as an initial demonstration of its function

That's being worked on in #5954 

> include languages in a dropdown

I thought about that but wanted to show the user's language up front so they would see something they recognised and click it. For example, if the default is English then the dropdown will show "English" (or "EN") which may not mean much to someone who speaks Swahili.

I took the Facebook login page for inspiration, which looks like this:

![Screenshot from 2020-03-17 08-58-05](https://user-images.githubusercontent.com/2064938/76795192-7b51d900-682d-11ea-9f13-d4cd5c43c089.png)

**Comment by michaelkohn** [comment]
Are we able to detect the language of the user's browser/device? It could be nice to automatically set the language of the app to the language that the user has already set on their browser or device (but still let them change it if they want).  If the language they have set in their browser or device doesn't exist in the app, set the app language t to English but let them choose others.

**Comment by garethbowen** [comment]
It is possible, but does not necessarily match one to one with the locale codes we use in the app. I think the default language for the project is likely to be a better default than the language configured on the device as most projects only fully support one language anyway.

**Comment by n-orlowski** [comment]
> For example, if the default is English then the dropdown will show "English" (or "EN") which may not mean much to someone who speaks Swahili.

Makes sense; my thoughts were that the default would be blank and that a user would need to open the dropdown to make a selection and recognize their language. Also not opposed to the list of languages all displayed but think they should look more like a selection instead of links (if the links don't change the language of the log in, similar to the Facebook example). 

I would still suggest placement above the button so the order of actions taken is chronological to the specific elements.

**Comment by garethbowen** [comment]
@n-orlowski Updated design, please let me know what you think!

Mobile

![Screenshot from 2020-03-18 11-49-29](https://user-images.githubusercontent.com/2064938/76908872-a829ed00-690e-11ea-86c6-01a5b1b16925.png)

Desktop

![Screenshot from 2020-03-18 11-49-44](https://user-images.githubusercontent.com/2064938/76908880-ac560a80-690e-11ea-97e8-9408f36ffc6f.png)

**Comment by n-orlowski** [comment]
Confirming I misunderstood this and the initial layout makes sense, sorry @garethbowen !

**Comment by michaelkohn** [comment]
Looks great!

**Comment by garethbowen** [comment]
Ready for AT in `6280-locale-selector-on-login`

**Comment by garethbowen** [comment]
This PR also implements the dynamic translation on the login page so if the user selects their language then the entire form will be immediately translated.

**Comment by ngaruko** [comment]
LGTM
One little thing @garethbowen, should this Bambara translation be shipped with the app or it is to be configured?
![image](https://user-images.githubusercontent.com/6979995/79285314-54acbe00-7f11-11ea-884d-0dbdb61025f7.png)

**Comment by ngaruko** [comment]
The login button translation seems also inconsistent :
*Not translated in Bambara
![image](https://user-images.githubusercontent.com/6979995/79285529-f3391f00-7f11-11ea-8ac8-7b54d059c48b.png 
*English translation for Bahasa, French and Spanish
![image](https://user-images.githubusercontent.com/6979995/79285575-1663ce80-7f12-11ea-85d9-0b1a906921db.png)
![image](https://user-images.githubusercontent.com/6979995/79285599-21b6fa00-7f12-11ea-8749-f90195de3d04.png)

![image](https://user-images.githubusercontent.com/6979995/79285641-401cf580-7f12-11ea-961c-8065cd43a076.png)

**Comment by garethbowen** [comment]
That's all configurable. Try setting the translations and make sure the login page is updated. NB: Make sure you clear your service worker cache fully before checking - until #6338 is done the cache doesn't clear automatically when config is changed.

**Comment by ngaruko** [comment]
Good to merge then @garethbowen ...
![image](https://user-images.githubusercontent.com/6979995/79290430-34383000-7f20-11ea-8385-22fc7126984e.png)
